packages/adapter-cloudflare/index.js

```diff
@@ -62,29 +62,53 @@ export default function () {
 }
 
 /**
- * @param {string} app_dir
+ * @param {import('@sveltejs/kit').Builder} builder
  * @param {string[]} assets
  * @returns {import('.').RoutesJSONSpec}
  */
-function get_routes_json(app_dir, assets) {
+function get_routes_json(builder, assets) {
+	/**
+	 * The list of routes that will _not_ invoke functions (which cost money).
+	 * This is done on a best-effort basis, as there is a limit of 100 rules
+	 */
+	const exclude = [`/${builder.config.kit.appDir}/*`];
+
+	const MAX_EXCLUSIONS = 98; // 100 minus existing include/exclude rules
+	let excess;
+
+	if (assets.length > MAX_EXCLUSIONS) {
+		excess = 'static assets';
+
+		if (builder.prerendered.paths.length > 0) {
+			excess = 'or prerendered routes';
+		}
+	} else if (assets.length + builder.prerendered.paths.length > MAX_EXCLUSIONS) {
+		excess = 'prerendered routes';
+	}
+
+	for (const file of assets) {
+		if (!file.startsWith(`${builder.config.kit.appDir}/`)) {
+			exclude.push(`/${file}`);
+		}
+	}
+
+	for (const path of builder.prerendered.paths) {
+		if (!builder.prerendered.redirects.has(path)) {
+			exclude.push(path);
+		}
+	}
+
+	if (excess) {
+		const message = `Static file count exceeds _routes.json limits (see https://developers.cloudflare.com/pages/platform/functions/routing/#limits). Accessing some files ${excess} will cause function invocations.`;
```

Comment:
it might read better if "files" were removed or changed to "files for"

Reply:
that was just a typo, `files` shouldn't be there at all